opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -15,78 +15,86 @@
-   * If the carrier object has no such span stored within it, a new Span is created.
-   *
-   * Unless there’s an error, it returns a SpanBuilder.
-   * The Span generated from the builder can be used in the host process like any other.
+  /**
+   * Extract a SpanContext from a `carrier` of a given type, presumably after propagation across a process boundary.
    *
-   * (Note that some OpenTracing implementations consider the Spans on either side of an RPC to have the same identity,
-   * and others consider the caller to be the parent and the receiver to be the child)
+   * <p>Example:
+   * <pre>{@code
+   * Tracer tracer = ...
+   * HttpHeaderReader httpHeaderReader = new AnHttpHeaderCarrier(httpRequest);
+   * SpanContext spanCtx = tracer.extract(httpHeaderReader);
+   * tracer.buildSpan('...').withChildOf(spanCtx).start();
+   * }</pre>
    *
-   * Attempting to join from a carrier that has been registered/configured to this Tracer will result in a
-   * IllegalStateException.
+   * If the span serialized state is invalid (corrupt, wrong version, etc) inside the carrier this will result in an IllegalArgumentException.
    *
-   * If the span serialized state is invalid (corrupt, wrong version, etc) inside the carrier this will result in a
-   * IllegalArgumentException.
+   * @param <C> the carrier type: all Tracer.extract() implementations must support io.opentracing.propagation.TextMapReader, io.opentracing.propagation.HttpHeaderReader, and java.nio.ByteBuffer
+   * @param carrier the carrier for the SpanContext state
+   * @returns the SpanContext instance extracted from the carrier
    *
-   * All implementations support at minimum the required carriers BinaryReader and TextMapReader.
+   * @see io.opentracing.propagation
    */
-  <T> SpanBuilder join(T carrier);
+  <C> SpanContext extract(C carrier);
 
 
   interface SpanBuilder {
 
-      /** Specify the operationName.
-       *
-       * If the operationName has already been set (implicitly or explicitly) an IllegalStateException will be thrown.
+      /**
+       * A shorthand for withReference(Reference.childOf(parent)).
        */
-      SpanBuilder withOperationName(String operationName);
+      SpanBuilder asChildOf(SpanContext parent);
 
-      /** Specify the parent span
+      /**
+       * Add a reference from the Span being built to a distinct (usually parent) Span. May be called multiple times to represent multiple such References.
+       *
+       * @param referenceType the reference type, typically one of the constants defined in References
+       * @param referencedContext the SpanContext being referenced; e.g., for a References.CHILD_OF referenceType, the referencedContext is the parent
        *
-       * If the parent has already been set an IllegalStateException will be thrown.
+       * @see io.opentracing.References
        */
-      SpanBuilder withParent(Span parent);
+      SpanBuilder addReference(Comparable referenceType, SpanContext referencedContext);
```

Comment:
if using Comparable, we should provide rationale for the choice, for example, why ordering matters. It is distracting otherwise.

A guess is that we are expecting an implementation to have multiple parents and need to know the parent vs adopted vs step-parent? I suppose we are expecting the implementation to retain the order?

Another note, Comparable has a type parameter `<T>` which would lead to this method having a compiler warning.